docs/sync_code.md

```diff
@@ -0,0 +1,34 @@
+# Basic workflow for Operator
+
+1. Create a branch:
+Start by creating a branch from "main" branch.
+
+2. Ensure local tests pass:
+Run unit and e2e tests, and ensure they pass before submitting a pull request (PR) targeting "main" branch.
+If the PR is associated with any Jira ticket, include the ticket link in the "Description" section of the PR.
+
+
+3. Review and merge PR:
+Once the PR has passed github checks and is reviewed and approved, the PR author should merge it into "main" branch.
```

Comment:
Is author really involved? Was it bot or maintainer?

Reply:
in my experience:
either i need to enable auto-merge myself or i need to manually click "merge" or someone helps to click "merge/auto-merge" for me, prow does not help in a 100% reliable way. 
until it is fully automated without so many "exceptions", i would prefer the author of the PR follow it till the end. 
if the author is a community user, then might have problem to "merge", maintainer can be tagged for such cases.  this could also happens when these PR need to get into downstream.